PyNvCodec/TC/src/Tasks.cpp

```diff
@@ -1284,3 +1313,173 @@ ResizeSurface* ResizeSurface::Make(uint32_t width, uint32_t height,
 {
   return new ResizeSurface(width, height, format, ctx, str);
 }
+
+// for RemapSurface
+namespace VPF
+{
+struct RemapSurface_Impl {
+  Surface* pSurface = nullptr;
+  SurfacePlane* xMapPlane = nullptr;
+  SurfacePlane* yMapPlane = nullptr;
+  CUcontext cu_ctx;
+  CUstream cu_str;
+  NppStreamContext nppCtx;
+
+  RemapSurface_Impl(const float* x_map, const float* y_map,
+                     uint32_t width, uint32_t height, Pixel_Format format,
+                     CUcontext ctx, CUstream str)
+      : cu_ctx(ctx), cu_str(str)
+  {
+    // init surface plane for map_x, map_y
+    xMapPlane = CreateSurfacePlaneByHostMemory(x_map, width, height, ctx, str);
+    yMapPlane = CreateSurfacePlaneByHostMemory(y_map, width, height, ctx, str);
+
+    SetupNppContext(cu_ctx, cu_str, nppCtx);
+  }
+
+  template <typename T>
+  static SurfacePlane* CreateSurfacePlaneByHostMemory(const T* p_src_host,
```

Comment:
Is it possible to use existing `CudaBuffer` class instead of `CreateSurfacePlaneByHostMemory` function?

`CudaBuffer` provides exactly same functionality (upload generic data to vRAM) and from Python side it supports numpy arrays as input.

If you change `RemapSurface::RemapSurface` constructor signature slightly to accept 2 `CudaBuffer` instead of raw `float*` pointers, it will be possible to keep `x_map` and `y_map` data on GPU (by uploading numpy array to GPU or obtaining remap data from other CUDA-enabled frameworks).

This is C++ code for `CudaBuffer` class: https://github.com/NVIDIA/VideoProcessingFramework/blob/ec453adcb2c3cd5cd566300d613d3ffd63343dda/PyNvCodec/TC/inc/MemoryInterfaces.hpp#L108

This is code which does mapping from C++ land to Python: https://github.com/NVIDIA/VideoProcessingFramework/blob/ec453adcb2c3cd5cd566300d613d3ffd63343dda/PyNvCodec/src/PyBufferUploader.cpp#L49

Please let me know if you need help with it.